saleor/graphql/account/utils.py

```diff
@@ -122,6 +124,30 @@ def get_out_of_scope_permissions(user: "User", permissions: List[str]):
     return missing_permissions
 
 
+def get_out_of_scope_users(root_user: "User", users: "QuerySet"):
+    """Return users whose permission scope is wider than the given user."""
+    out_of_scope_users = []
+    root_user_permission_pks = set(
+        get_user_permissions(root_user).values_list("pk", flat=True)
+    )
+
+    users_data = users.annotate(
+        group_perms=ArrayAgg("groups__permissions"),
+        user_perms=ArrayAgg("user_permissions__pk"),
+    ).values("group_perms", "user_perms")
+
+    for index, user_data in enumerate(users_data):
+        user_permissions = set(user_data["group_perms"]).union(
+            set(user_data["user_perms"])
+        )
+        user_permissions.discard(None)
```

Comment:
Shouldn't we somehow use `get_user_permissions` here? One problem that I see is that `get_user_permission` always performs database query so this might inefficient.

Reply:
Because of that, I decided to not use it for every user which we need to check. When it will be more users there will be plenty of database queries.